index.js

```diff
@@ -2120,24 +2124,41 @@
   //. Cons('red', Cons('blue', Cons('green', Nil)))
   //. ```
   function sortBy(f, foldable) {
-    var rs = reduce(function(xs, x) {
-      var fx = f(x);
-      var lower = 0;
-      var upper = xs.length;
-      while (lower < upper) {
-        var idx = Math.floor((lower + upper) / 2);
-        if (lte(xs[idx].fx, fx)) lower = idx + 1; else upper = idx;
-      }
-      xs.splice(lower, 0, {x: x, fx: fx});
+    var foldableIsArray = Array.isArray(foldable);
+    var idx = 0;
+    function reducer(xs, x) {
+      xs.push({idx: idx, x: x, prop: f(x)});
+      idx += 1;
       return xs;
-    }, [], foldable);
-
-    var F = foldable.constructor;
-    var result = empty(F);
-    for (var idx = 0; idx < rs.length; idx += 1) {
-      result = concat(result, of(F, rs[idx].x));
     }
-    return result;
+    var rs = reduce(reducer, [], foldable);
+    if (rs.length === 0) {
+      return foldable;
+    }
+    var lteFn =
+      typeof rs[0].prop === 'string' || typeof rs[0].prop === 'number' ?
+        function(a, b) { return a <= b; } :
+        lte;
+    rs.sort(function(a, b) {
+      return lteFn(a.prop, b.prop) ?
+        lteFn(b.prop, a.prop) ? a.idx - b.idx : -1 :
+        1;
+    });
+    if (foldableIsArray) {
```

Comment:
Since `foldableIsArray` is only referenced here, let's replace this identifier with `Array.isArray(foldable)`.

Reply:
Good idea. I was using it in several places earlier but that went away.

Reply:
:zap: